constraint/pkg/client/drivers/k8scel/driver.go

```diff
@@ -291,11 +285,12 @@ func assumeVAPEnforcementWithDefault(obj runtime.Object, vapDef vapDefault) (boo
 	}
 	switch vapDefault(shouldGen) {
 	case VAPDefaultYes:
-		return true, nil
+		return true
 	case VAPDefaultNo:
-		return false, nil
+		return false
+	// on unrecognized value, use the default
 	default:
-		return false, fmt.Errorf("unknown VAP generation label value, %q, should be \"yes\" or \"no\"", shouldGen)
+		return vapDef == VAPDefaultYes
```

Comment:
In GK, when label is not set to yes or no, we default to parent/CT value, then should this return VAPDefaultYes?

Reply:
This is logically equivalent.

If evaluating for a constraint, the call is:

`assumeVAPEnforcementNotDisabled := assumeVAPEnforcementWithDefault(constraint, VAPDefaultYes)`

For a constraint:

* If label is no, return false
* if label is yes/unknown, return true (vapDef == VAPDefaultYes)

For a constraint template

* If no default is provided, assume VAP is disabled, return false
* If label is no, return false
* If label is yes, return true
* If label is unknown, return according to user-provided default (yes => true, no => false)

Line 199: `if isAdmission && assumeVAPEnforcementNotDisabled && wrappedValidator.assumeVAPEnforcement {`

performs the aggregation:

* if not admission,  do not assume VAP handles enforcement (VAP only works for admission) else
* if the constraint has given an explicit "No" (defined by "for a constraint" above returning false), do not assume VAP handles enforcement
* if the constraint template test assumes no VAP enforcement (as defined by "for a constraint template" returning false), do not assume VAP handles enforcement.

TL;DR constraints can only disable enforcement with an explicit "No"

Reply:
> For a constraint:
...
if label is yes/unknown, return true (vapDef == VAPDefaultYes)

ok vapDef is aways VAPDefaultYes for constraint because of 
https://github.com/open-policy-agent/frameworks/pull/400/files#diff-5e2ac34b20b681ef5a8747fac1f9563e8c60060f9609e7ee5d10f63a894fd251R195

Reply:
Yep!